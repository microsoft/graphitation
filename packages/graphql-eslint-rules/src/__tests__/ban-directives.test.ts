import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../ban-directives";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("ban-directives", rule, {
  valid: [
    {
      code: `
query myOperation {
  user
  description
}`,
      options: [{ bannedDirectives: ["client"] }],
    },
    {
      code: `
query myOperation {
  user @foobar
  description
}`,
      options: [{ bannedDirectives: ["client"] }],
    },
  ],
  invalid: [
    {
      code: `
query myOperation {
  user @client
  description @foobar
}`,
      options: [{ bannedDirectives: ["client"] }],
      errors: 1,
    },
    {
      code: `
query myOperation {
  user @client
  description @foobar
}`,
      options: [{ bannedDirectives: ["client", "foobar"] }],
      errors: 2,
    },
  ],
});
