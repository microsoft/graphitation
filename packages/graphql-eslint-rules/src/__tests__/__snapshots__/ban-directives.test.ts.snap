// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
"  1 |
  2 | query myOperation {
> 3 |   user @client
    |         ^ Use of directive @client is prohibited
  4 |   description @foobar
  5 | }"
`;

exports[` 2`] = `
"  1 |
  2 | query myOperation {
> 3 |   user @client
    |         ^ Use of directive @client is prohibited
  4 |   description @foobar
  5 | }"
`;

exports[` 3`] = `
"  1 |
  2 | query myOperation {
  3 |   user @client
> 4 |   description @foobar
    |                ^ Use of directive @foobar is prohibited
  5 | }"
`;
